server/src/main/java/org/elasticsearch/common/Rounding.java

```diff
@@ -74,7 +75,8 @@ long extraLocalOffsetLookup() {
 
             @Override
             long roundFloor(long utcMillis, int multiplier) {
-                return multiplier == 1 ? DateUtils.roundYear(utcMillis) : DateUtils.roundYearInterval(utcMillis, multiplier);
+                assert multiplier == 1;
+                return DateUtils.roundYear(utcMillis);
```

Comment:
Please confirm my understanding that this check was showing in profile?

Reply:
Not that one specifically, but the equivalent for `WEEK_OF_YEAR`, in particular I see the cost of `roundWeekIntervalOfWeekYear()`. It's smaller than the slowdown we are seeing in the nightlies though.
My doubt is that these methods are really small, so they are probably hard to sample and catch in flamegraphs. 
On the other hand, in the [commit diff](https://github.com/elastic/elasticsearch/compare/47edd0adb858e738b177c8b7e8564989415564ab...9f6eb1d4e3dbb4fdc10f9807d11c37047cd1da2f) there is really nothing apart from this change (all the rest is tests), so this is my only candidate...

Reply:
> My doubt is that these methods are really small, so they are probably hard to sample and catch in flamegraphs.

If they're small but called a lot, I would statistically expect them to still show up.